multibase.go

```diff
@@ -121,3 +133,36 @@ func Decode(data string) (Encoding, []byte, error) {
 		return -1, nil, ErrUnsupportedEncoding
 	}
 }
+
+var base32StdLowerPad = b32.NewEncodingCI("abcdefghijklmnopqrstuvwxyz234567")
+var base32StdLowerNoPad = base32StdLowerPad.WithPadding(b32.NoPadding)
+
+var base32StdUpperPad = b32.NewEncodingCI("ABCDEFGHIJKLMNOPQRSTUVWXYZ234567")
+var base32StdUpperNoPad = base32StdUpperPad.WithPadding(b32.NoPadding)
+
+var base32HexLowerPad = b32.NewEncodingCI("0123456789abcdefghijklmnopqrstuv")
+var base32HexLowerNoPad = base32HexLowerPad.WithPadding(b32.NoPadding)
+
+var base32HexUpperPad = b32.NewEncodingCI("0123456789ABCDEFGHIJKLMNOPQRSTUV")
+var base32HexUpperNoPad = base32HexUpperPad.WithPadding(b32.NoPadding)
+
+
+func hexEncodeToStringUpper(src []byte) string {
```

Comment:
I'm not a fan of embedding all this stuff in this repo but I'm not sure of a better way to do this (without massive over-engineering and making things needlessly complicated).

We should at least move the base32/base16 functions to their own files.

Reply:
Agreed.  Will do.

Reply:
And done.